tests/loaders/test_mqtt.py

```diff
@@ -68,21 +69,10 @@ def read(self, *args) -> MockSeekMessage | None:
 
 
 @pytest.fixture
-def mock_paho(monkeypatch: pytest.MonkeyPatch) -> Iterator[MagicMock]:
-    with monkeypatch.context() as m:
-        mock_paho = MagicMock()
-        m.setitem(sys.modules, "paho", mock_paho)
-        m.setitem(sys.modules, "paho.mqtt", mock_paho.mqtt)
-        m.setitem(sys.modules, "paho.mqtt.client", mock_paho.mqtt.client)
+def mock_client(monkeypatch: pytest.MonkeyPatch) -> None:
+    import paho.mqtt.client as mqtt
 
-        yield mock_paho
-
-
-@pytest.fixture
-def mock_client(mock_paho: MagicMock) -> Iterator[MagicMock]:
-    mock_client = MQTTMock()
-    mock_paho.mqtt.client.Client.return_value = mock_client
-    yield mock_client
+    monkeypatch.setattr(mqtt, "Client", MQTTMock)
```

Comment:
For reviewer: this seems simpler,  since we do not yet need to manipulate the mock instance of `MQTTMock` inside the test.

The other approach also required us to do our imports from `dissect.target.loaders.mqtt` inside the test.